src/Microsoft.Health.Core/Features/Events/EventGridPublisher.cs

```diff
@@ -0,0 +1,78 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core.Pipeline;
+using Azure.Messaging.EventGrid;
+using EnsureThat;
+using Microsoft.Health.Abstractions.Features.Events;
+
+namespace Microsoft.Health.Core.Features.Events
+{
+    /// <summary>
+    /// EventGridPublisher.
+    /// </summary>
+    public class EventGridPublisher : IEventGridPublisher
+    {
+        private readonly EventGridPublisherClient _client;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventGridPublisher"/> class.
+        /// </summary>
+        /// <param name="endpoint">Uri of topic</param>
+        /// <param name="key">access key</param>
+        public EventGridPublisher(Uri endpoint, string key)
```

Comment:
What's the intended lifetime of this? Is it going to be a `singleton` or `scoped` or used by a factory? The reason I ask is it may inform how we want to handle this what I presume is a secret that will need to be rotated at some point.

Reply:
Yes it would be a singleton injected later. We don't need this access key option now, given the PaaS would only use the client certificate option. But I left this in here so if someone wants to use on the OSS side with just the access key they still could.